### ISSUE TITLE
Pseudotiled windows interact oddly with max-layout frames.

### ISSUE THREAD

**Issue by gravndal** [herbstluftwm/herbstluftwm#1392]
![2021-12-05-08:59:23](https://user-images.githubusercontent.com/4656860/144738594-5f3b8a7a-e6b5-46b9-925b-bd39cde9fc8b.png)

This is three terminals in one frame, the focus window is pseudotiled.

Pseudotiling probably doesn't see that much use what with floating windows being supported, but this genuinely confused me for a few seconds when I toggle floating mode (`hc floating`) as it looks like two monitors stacked on top of each other.

What would be the most "correct" way of drawing window decorations here is probably debatable.

**Comment by t-wissmann** [comment]
Yes, I also wondered what the best way is. The main question is:

* should the pseudotiled client "188" be listed as a tab in the decoration of client "187"? (in you screenshot)

If it is shown, then you see the tab for it twice. If it is not shown, then I think the tab should still be shown if the pseudotiled client is not visible (because otherwise, one easily looses the pseudotiled client because it does not have a tab). But this would then imply that the number of tabs varies between 2 and 3 in your example, depending on whether the pseudotiled client is shown or covered.

So in total I decided to go for the most honest way: in a frame in max layout, the tab bar shows all clients of this frame.

**Comment by t-wissmann** [comment]
Since I don't see a better way to draw tabs here, I think the behaviour should remain this way.
A bit of redundancy isn't bad for the sake of clarity.
If anybody has a better proposal, which tabs to display where, feel free to re-open.

**Comment by reduakt** [comment]
This issue and the current placement of the title bars tickles my eyes a lot. I think a duplicated title bar looks too cluttered.

How about hiding the title bar of the pseudo tiled window? Maybe apply something like title_when=none setting only to the pseudo tiled window.
Just the upper (main) title bar is sufficient. By cycling through the windows, we should `cycle +1` through this upper title list and select and highlight the entries only on this upper bar.

A hidden pseudotiled window would be treated like a tab which is in the background. You need to cycle through the upper bar to see it. 

The approach should look like this:

<img src="https://user-images.githubusercontent.com/12463729/162417139-d0ce3f35-b47a-4896-908f-a8aa7ff0e96b.png" width="300" >

**Comment by t-wissmann** [comment]
I'm not entirely convinced, mainly because I don't see the need for pseudotile anyway, given that hlwm now supports single-window floating. Concretely: why don't you make the "Application Finder" window floating?

If you still insist on having it pseudotiled, I can implement it (implementation note to myself: do this in the same loop when setting the visibility of the windows in the max frame).

There might be an odd corner case: assume you have two pseudotiled windows, where the focused one is larger then the one below. Then it might make sense to have tab bars on both of them. (But I don't think that it is worth the time to take care of this corner case in the implementation).
![image](https://user-images.githubusercontent.com/9048813/162614233-e1b22ac3-9cc0-4e10-ab62-1d8fa111379c.png)

**Comment by reduakt** [comment]
Pseudotiling is a nice feature to keep small your windows structured and organized.
Floating means mouse interaction to move or resize. To my mind, floating should be used as a fallback, but not as a default.

My current `layout.cpp `:
```
@@ -256,7 +256,9 @@ TilingResult FrameLeaf::layoutMax(Rectangle rect) {
             step.needsRaise = true;
         }
         if (settings_->tabbed_max()) {
-            step.tabs = clients;
+            if (!client->pseudotile_()) {
+                step.tabs = clients;
+            }
         }
         res.add(client, step);
     }
```
Any ideas how to improve? I dig through the code to get a better understanding.

**Comment by t-wissmann** [comment]
> Floating means mouse interaction to move or resize.

Not really. You can move and resize floating windows with the same keybindings that you also use for tiled windows.

Your code just disables tab bars for all pseudotiled windows. I had a more complicated strategy in mind, which would result in the bottom-most still visible client having the tab bar. But of course this would result in the tab bar jumping around when switching between two pseudotiled windows in the same max-frame.

Also, in your screenshot, the "Application finder" tab has the active color, but I find this very misleading because it indicates that the tiled window was the "Application finder".

**Comment by reduakt** [comment]
> Not really. You can move and resize floating windows with the same keybindings that you also use for tiled windows.
thanks for the hint, i tried it. 

> Your code just disables tab bars for all pseudotiled windows. I had a more complicated strategy in mind, which would result in the bottom-most still visible client having the tab bar. But of course this would result in the tab bar jumping around when switching between two pseudotiled windows in the same max-frame.
My code disables all other clients listed in the title bar of the pseudotiled window. It is on step in the direction to avoid two title bars with identical entries of every client from this frame.

> Also, in your screenshot, the "Application finder" tab has the active color, but I find this very misleading because it indicates that the tiled window was the "Application finder".
The created image should explain, that a pseudotiled window, which comes from a tabbed max-layout design, should not create its own framedecoration (tabbar). The pseudotiled window should act like a normal tabbed maxlayout window, except for the outer client borders tight around the application window itself.

I can try to explain via IRC to make my approach more clear.
